Framework/DataHandling/inc/MantidDataHandling/MaskDetectors.h

```diff
@@ -80,14 +80,28 @@ class DLLExport MaskDetectors : public API::Algorithm {
   void init() override;
   void exec() override;
   void execPeaks(DataObjects::PeaksWorkspace_sptr WS);
-  void fillIndexListFromSpectra(std::vector<size_t> &indexList,
-                                const std::vector<specnum_t> &spectraList,
-                                const API::MatrixWorkspace_sptr WS);
-  void appendToIndexListFromWS(std::vector<size_t> &indexList,
-                               const API::MatrixWorkspace_sptr maskedWorkspace);
+  void
+  fillIndexListFromSpectra(std::vector<size_t> &indexList,
+                           const std::vector<specnum_t> &spectraList,
+                           const API::MatrixWorkspace_sptr WS,
+                           const std::tuple<size_t, size_t, bool> &range_info);
```

Comment:
`std::tuple` is an interesting choice here. Why not use a plain old struct instead? IMHO, it is easier to read
```
range_info.min = 0;
```
than
```
std::get<0>(range_info) = 0;
```
no?